plugins/dss/hooks.js

```diff
@@ -15,10 +15,196 @@ import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
 import FormGroupHeadingContent
   from "#SRC/js/components/form/FormGroupHeadingContent";
 import FormRow from "#SRC/js/components/form/FormRow";
+import VolumeConstants
+  from "#PLUGINS/services/src/js/constants/VolumeConstants";
+import {
+  getContainerNameWithIcon
+} from "#PLUGINS/services/src/js/utils/ServiceConfigDisplayUtil";
+import { omit } from "#SRC/js/utils/Util";
 
 const errorsLens = Objektiv.attr("container", {}).attr("volumes", []);
 
-function getVolumeTypes(props) {
+function getContainerMounts(containers, volumeMountIndex, volumeMounts) {
+  return containers.map((container, containerIndex) => {
+    let containersLabel = null;
+    let pathLabel = null;
+    if (containerIndex === 0) {
+      containersLabel = (
+        <FieldLabel>
+          <FormGroupHeading>
+            <FormGroupHeadingContent primary={true}>
+              Containers
+            </FormGroupHeadingContent>
+          </FormGroupHeading>
+        </FieldLabel>
+      );
+      pathLabel = (
+        <FieldLabel>
+          <FormGroupHeading>
+            <FormGroupHeadingContent primary={true}>
+              Container Path
+            </FormGroupHeadingContent>
+          </FormGroupHeading>
+        </FieldLabel>
+      );
+    }
+
+    return (
+      <FormRow key={containerIndex}>
+        <FormGroup className="column-3">
+          {containersLabel}
+          <div className="form-control-input-height">
+            {getContainerNameWithIcon(container)}
+          </div>
+        </FormGroup>
+        <FormGroup className="column-9">
+          {pathLabel}
+          <FieldInput
+            name={`volumeMounts.${volumeMountIndex}.mountPath.${containerIndex}`}
+            type="text"
+            value={volumeMounts[volumeMountIndex].mountPath[containerIndex]}
+          />
+        </FormGroup>
+      </FormRow>
+    );
+  });
+}
+
+function VolumeMountTypesSelect(props) {
+  return (
+    <FieldSelect
+      name={`volumeMounts.${props.index}.type`}
+      value={props.volumes.type}
+    >
+      <option value="">Select...</option>
+      <option value={VolumeConstants.type.host}>Host Volume</option>
+      <option value={VolumeConstants.type.ephemeral}>
+        Ephemeral Volume
+      </option>
+      <option value={VolumeConstants.type.localPersistent}>
+        Local Persistent Volume
+      </option>
+      <option value={VolumeConstants.type.dss}>
+        DC/OS Storage Service
+      </option>
+    </FieldSelect>
+  );
+}
+
+function DSSInput(props) {
+  const { volumes, index, offset } = props;
+  if (volumes.type !== VolumeConstants.type.dss) {
+    return (
+      <div>
+        <FieldLabel>
+          Unable to edit this Volume
```

Comment:
Just btw, VolumesFormSection puts a space after this, should probably remove there too for consistent: https://github.com/dcos/dcos-ui/blob/master/plugins/services/src/js/components/forms/VolumesFormSection.js#L296

Reply:
Yes you are right I will do that in another PR.